### ISSUE TITLE
Unable to fork Sandboxes

### ISSUE THREAD

**Issue by twosavoie** [codesandbox/codesandbox-client#8344]
# 🐛 bug report

## Preflight Checklist

- [x] I have read the [Contributing Guidelines][contributing] for this project.
- [x] I agree to follow the [Code of Conduct][code_of_conduct] that this project
      adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

## Desciption of the problem
Update: a sandbox can be forked by clicking the icon next to "Fork" and then choose a username. I feel like that's a small target to hit. And a person would expect that clicking "Fork" would fork the repo. 

I am unable to fork a sandbox that I don't own. I am able to fork my own sandboxs. When I try to fork, I receive a 401 error. "Forking Failed (401) You don't have the needed authorization for this workspace (skillcrush-curriculum)"

Before this, I was able to fork without issue.

Thanks for your help!

## How has this issue affected you? What are you trying to accomplish?
I'm trying to fork a sandbox to complete a lesson. 

### To Reproduce
Try to fork a sandbox you don't own.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      |              |
| Browser          |     Chrome         |
| Operating System |      Mac/Sonoma        |

[code_of_conduct]: https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md
[contributing]: https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md

**Comment by JamesACS** [comment]
Hello @twosavoie, 

In our workspace settings we have a 'Prevent leaving workspace' toggle that owners can enable, which seems to have been enabled by the owner of this environment. 

We'd recommend contacting them directly to ask them to toggle this off.

**Comment by twosavoie** [comment]
Hi, thanks for getting back to me so quickly!

I've checked the settings on the owner's account for the sandbox itself and Permissions and the "Prevent Leaving Workspace" option hasn't been toggled on. (see below) Is there a different place I should look?

Overnight I saw that clicking the word "Fork" created the error, but clicking the icon to the right of "Fork" and then the username in the dropdown allowed the sandbox to be forked. 

Thanks again for your help. 

![Screenshot 2024-02-14 at 10 03 42 AM (2)](https://github.com/codesandbox/codesandbox-client/assets/35409111/05b2eb37-c4cb-4f57-9009-1aa5089ec1e3)
![Screenshot 2024-02-14 at 10 05 01 AM (2)](https://github.com/codesandbox/codesandbox-client/assets/35409111/5c68a7a9-dbcb-498c-b27c-a865651d2869)
